kimu/core/intersection_tool_line_circle.py

```diff
@@ -0,0 +1,272 @@
+        if self.iface.activeLayer() != self.layer:
+            LOGGER.warning(tr("Please select a line layer"), extra={"details": ""})
+            return
+
+        if len(self.iface.activeLayer().selectedFeatures()) != 1:
+            LOGGER.warning(tr("Please select only one line"), extra={"details": ""})
+            return
+        else:
+            geometry = self.iface.activeLayer().selectedFeatures()[0].geometry()
+
+        m = self.iface.mapCanvas().snappingUtils().snapToMap(event.pos())
+        self.i.setMatch(m)
+
+        # Coordinates of the point to be used as a circle centroid
+        x_coord = event.pos().x()
+        y_coord = event.pos().y()
+        point = self.canvas.getCoordinateTransform().toMapCoordinates(x_coord, y_coord)
+        centroid = [point[0], point[1]]
+
+        self._intersect(geometry, centroid)
+
+    def _intersect(self, geometry: QgsGeometry, centroid: List) -> QgsVectorLayer:
+        """Determine intersection point(s) of the selected
+        line and implicitly determined circle."""
+        result_layer1 = QgsVectorLayer("Point", "temp", "memory")
+        crs = self.layer.crs()
+        result_layer1.setCrs(crs)
+        result_layer1_dataprovider = result_layer1.dataProvider()
+        result_layer1_dataprovider.addAttributes(
+            [QgsField("xcoord", QVariant.Double), QgsField("ycoord", QVariant.Double)]
+        )
+        result_layer1.updateFields()
+
+        result_layer2 = QgsVectorLayer("Point", "temp", "memory")
+        crs = self.layer.crs()
+        result_layer2.setCrs(crs)
+        result_layer2_dataprovider = result_layer2.dataProvider()
+        result_layer2_dataprovider.addAttributes(
+            [QgsField("xcoord", QVariant.Double), QgsField("ycoord", QVariant.Double)]
+        )
+        result_layer2.updateFields()
+
+        line_feat = geometry.asPolyline()
+        start_point = QgsPointXY(line_feat[0])
+        end_point = QgsPointXY(line_feat[-1])
+        line_coords = [start_point.x(), start_point.y(), end_point.x(), end_point.y()]
+
+        # In crs units (meters for EPSG: 3067)
+        r = self.ui.get_radius()
+
+        # DUMMY TEST
+        # x_sol1 = float(centroid[0]) + r * 100.0
+        # y_sol1 = float(line_coords[3]) + r * 100.0
+        # x_sol2 = float(centroid[0]) - r * 100.0
+        # y_sol2 = float(line_coords[3]) - r * 100.0
+
+        # Which coordinate value is stored at which index of the list
+        # x1 = line_coords[0]
+        # y1 = line_coords[1]
+        # x2 = line_coords[2]
+        # y2 = line_coords[3]
+
+        # Debug tests
+        # print(f"Value of line_coords[3] is {line_coords[3]}")
+        # aa = (line_coords[3])**2.0
+        # print(f"Value of aa is {aa}")
+        # print(f"Value of r is {r}")
+
+        # Determine the intersection point with the help of analytical geometry
+        # 1. Determine the function of the straight line the selected
```

Comment:
Steps 1,2,3 probably represent a,b,c below? Maybe move comment steps 2 and 3 closer to b and c.

Reply:
Actually the steps do not represent a, b and c. The steps try to give a general idea on what I did in order to figure out the equations used for calculating the values for a, b and c. Parameters a, b and c, in turn, are the a, b and c from https://en.wikipedia.org/wiki/Quadratic_equation#Quadratic_formula_and_its_derivation

Reply:
Documentation of this phase will be improved in the following commits to make it more clear what is going on here